sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/HDFSFileSink.java

```diff
@@ -209,7 +209,12 @@
       if (configuration == null) {
         configuration = new Configuration(false);
       }
-      return this.setSerializableConfiguration(new SerializableConfiguration(configuration));
+
+      SerializableConfiguration serializableConfiguration =
+        new SerializableConfiguration(configuration);
+      SerializableConfiguration.readConfigurationFromFile(serializableConfiguration);
```

Comment:
Set a value by `set` method will not overwrite by `addResource` regardless of call order, so when I have a file named `test.xml`
```
<configuration>
    <property>
        <name>color</name>
        <value>yellow</value>
    </property>
</configuration>
```
then output of 
```
Configuration conf = new Configuration();
conf.addResource("file://test.xml");
conf.set("color", "will not overwrite");
System.out.println(conf.get("color"));
```
and
```
Configuration conf = new Configuration();
conf.set("color", "will not overwrite");
conf.addResource("file://test.xml");
System.out.println(conf.get("color"));
```
is same.